File: selectionsort.c

#include <stdio.h>

int main()
{

int a[] = {3, 2, 6, 4, 9, 7};
int a_size;
int i,j,temp;
//temp = a[0];
a_size = sizeof(a)/sizeof(int);


for(j=0;j<(a_size);j++)
{

int location = a[j];

for(i=j+1;i<(a_size);i++)
{
   if(location < a[i])
   {
      continue;
   }
   else if(location > a[i])
   {
      location = a[i];
   }
  //if swap func exists, call here
  temp = a[j];
  a[j] = location;
  a[i] = temp;
}

  
}

for ( int c=0; c < a_size; c++)
{
  printf("%d",a[c]);
}



}

File: selectionsort.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Thu Feb 13 14:16:20 2020

@author: joncauchi
"""


list = [3,2,7,5,9,6,1]
 
for i in range(len(list)):
    
    location = i
    for j in range(i+1, len(list)):
        
        if(list[location] > list[j]):
            location = j
        
        elif(list[location]< list[j]):
            pass #do nothing
            
    #swap
    temp = list[i]
    list[i] = list[location]
    list[location] = temp
    
for c in list:
    print(c)